tests/nnstreamer_filter_llamacpp/unittest_filter_llamacpp.cc

```diff
@@ -0,0 +1,372 @@
+/**
+ * @file	unittest_filter_custom.cc
+ * @date	23 May 2025
+ * @brief	Unit test for tensor filter llamacpp plugin
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Hyunil Park <hyunil46.park@samsung.com>
+ * @bug		No known bugs.
+ */
+
+#include <gtest/gtest.h>
+#include <gst/app/gstappsink.h>
+#include <gst/app/gstappsrc.h>
+#include <gst/gst.h>
+#include <unittest_util.h>
+
+
+/* Static variable to track the number of async callback invocations. */
+static int new_sample_count = 0;
+
+/**
+ * @brief Callback function for appsink element to pull sample and print the output text from llamacpp plugin
+ */
+static GstFlowReturn
+new_sample_cb (GstElement *sink, gpointer user_data)
+{
+  GstSample *sample;
+  GstBuffer *buffer;
+  GstMapInfo map;
+
+  new_sample_count++;
+  g_signal_emit_by_name (sink, "pull-sample", &sample);
```

Comment:
suggestion:
sample = gst_app_sink_pull_sample (GST_APP_SINK (sink));

Reply:
Thank you. I've updated it.